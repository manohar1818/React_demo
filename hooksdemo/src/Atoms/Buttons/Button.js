import React, { useState } from 'react';


const Example = ({ text }) => {
    return (
      <button>
        {text}
      </button>
    );
  };

export default Example