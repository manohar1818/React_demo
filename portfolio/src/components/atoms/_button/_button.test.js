import React from 'react';
import { create } from "react-test-renderer";
import { render, fireEvent, getByRole } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from './_button';

describe("Button component",() =>{
    test("Matches the snapshot",() => {
        const button = create(<Button />);
        expect(button.toJSON()).toMatchSnapshot();

    })
    test("button element is rendered",() => {
        const { getByTestId, getByText } = render(<Button />);
        expect(getByTestId("button")).toBeInTheDocument();
        
    });
    test("button element is enabled",() => {
        const { getByTestId, getByText } = render(<Button value="save" disabled={false}/>);
        expect(getByTestId("button")).toBeEnabled();
        
    });
    test("button element is disabled",() => {
        const { getByTestId, getByText } = render(<Button value="save" disabled={true}/>);
        expect(getByTestId("button")).toBeDisabled();
        
    });
   
    test("on click of button",() => {
        const mockOnClick = jest.fn();
        const { getByTestId, getByText,getByRole } = render(<Button value="save" disabled={false} onClick={mockOnClick}/>);
        expect(getByTestId("button")).toBeEnabled();
        fireEvent.click(getByRole("button"));
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
});