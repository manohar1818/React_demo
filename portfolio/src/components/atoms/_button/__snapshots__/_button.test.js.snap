// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="makeStyles-button-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-testid="button"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        className="MuiTypography-root MuiTypography-button"
      >
         
      </span>
    </span>
  </button>
</div>
`;
